### ISSUE TITLE
Turn CI runs as PR checks for merging

### ISSUE THREAD

**Issue by juanmanso** [masa-finance/masa-bittensor#108]
## Description

After [tbd] and [tbd], our codebase is free of linting issues and CI is passing. So this task is for repo admins to turn on CI check for PRs

## Accpetance criteria
- Failing CI runs block merge of PRs
- Properly formatted and linted code should not be blocked by CI to merge its PR

## Related issues

- Related to #103
- Related to #105
- Blocked by #106
- Blocked by #107

**Comment by Luka-Loncar** [comment]
Wontfix
